presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -588,113 +564,91 @@ public void addInt(int value)
         }
     }
 
-    public class ParquetJsonColumnConverter
+    public class ParquetColumnConverter
             extends GroupConverter
-            implements Closeable
     {
-        private final DynamicSliceOutput out = new DynamicSliceOutput(1024);
-
-        private final GroupedJsonConverter jsonConverter;
+        private final GroupedConverter groupedConverter;
         private final int fieldIndex;
 
-        private JsonGenerator generator;
+        private ArrayBuilder builder;
 
-        public ParquetJsonColumnConverter(GroupedJsonConverter jsonConverter, int fieldIndex)
+        public ParquetColumnConverter(GroupedConverter groupedConverter, int fieldIndex)
         {
-            this.jsonConverter = jsonConverter;
+            this.groupedConverter = groupedConverter;
             this.fieldIndex = fieldIndex;
         }
 
         @Override
         public Converter getConverter(int fieldIndex)
         {
-            return jsonConverter.getConverter(fieldIndex);
+            return groupedConverter.getConverter(fieldIndex);
         }
 
         @Override
         public void start()
         {
-            try {
-                out.reset();
-                generator = new JsonFactory().createGenerator(out);
-            }
-            catch (IOException e) {
-                throw Throwables.propagate(e);
-            }
+            builder = ArrayBuilder.builder();
 
-            jsonConverter.beforeValue(generator);
-            jsonConverter.start();
+            groupedConverter.beforeValue(builder);
+            groupedConverter.start();
         }
 
         @Override
         public void end()
         {
-            jsonConverter.end();
-            jsonConverter.afterValue();
+            groupedConverter.end();
+            groupedConverter.afterValue();
 
             nulls[fieldIndex] = false;
-            try {
-                generator.close();
-            }
-            catch (IOException e) {
-                throw Throwables.propagate(e);
-            }
-            slices[fieldIndex] = out.copySlice();
-        }
-
-        @Override
-        public void close()
-                throws IOException
-        {
-            Closeables.closeQuietly(jsonConverter);
+            slices[fieldIndex] = builder.getSlice(0);
```

Comment:
Shouldn't this be builder.toSlice()? Otherwise I think you'll end up with only the first element, since of the whole array